mesonbuild/backend/ninjabackend.py

```diff
@@ -2282,7 +2282,7 @@ def get_user_option_args(self):
     def generate_utils(self, outfile):
         cmd = [sys.executable, self.environment.get_build_command(),
                '--internal', 'scanbuild', self.environment.source_dir, self.environment.build_dir,
-               sys.executable, self.environment.get_build_command()] + self.get_user_option_args()
+               sys.executable, self.environment.get_build_command()] + sorted(self.get_user_option_args())
```

Comment:
I went through the code and the simplest solution seems to be to move the sorting to `get_user_option_args`. Just add a comment stating that the ordering must be the same for reproducibility reasons and mention that the order does not really matter, because the compiler flags produced do not clash with each other.